core/interception.js

```diff
@@ -145,340 +172,446 @@ wsHook.after = function(messageEvent, url)
+                                            {
+                                                console.log("WhatsIncognito: Error saving msg, msg ID already exists");
+                                            } else
+                                            {
+                                                console.log("WhatsIncognito: Unexpected error saving deleted msg")
+                                            }
+                                        };
+                                        request.onsuccess = (e) =>
+                                        {
+                                            console.log("WhatsIncognito: Saved deleted msg with ID " + deletedMsgContents.id + " from " + deletedMsgContents.from + " successfully.")
+                                        }
+                                    }
+                                    else
+                                    {
+                                        console.log("WhatsIncognito: Deleted msg contents not found")
+                                    }
+
+                                }
+                                runMe()
+                                break
+                            }
+                        }
+                    }
 
-            var message = parseMessage(children[i]);
-            if (message) messages.push(message);
+                    console.log("WhatsIncognito: --- Blocking message REVOKE action! ---");
+                    return false;
+                }
+            }
+        }
+        catch (exception)
+        {
+            console.error("WhatsIncognito: Allowing WA packet due to exception:");
+            console.error(exception);
+            return true;
         }
-    }
 
-    if ("search" === type)
-    {
-        messages = {eof: "true" === nodeReader.attr("last", node), messages: messages};
+        return true;
     }
-        
-    if (messages.length > 0)
+
+    NodeHandler.manipulateReceivedNode = function (node)
     {
-        if (WAdebugMode) console.log("Got messages! (count: " + messages.length +" )");
-    
-        if (isScrappingMessages)
+        var messages = [];
+        var children = nodeReader.children(node);
+        var tag = nodeReader.tag(node);
+        var type = nodeReader.attr("type", node);
+
+        if (Array.isArray(children))
         {
-            isScrappingMessages = false;
-            messages = messages.concat(messages);
+            for (var i = 0; i < children.length; i++)
+            {
+                var child = children[i];
+                var action = child[0];
+                var data = child[1];
 
-            if (WAdebugMode) console.log(JSON.parse(JSON.stringify(messages)));
-            //handler.scrapMessages(t[0].key.remoteJid, t[0].key.id, 50);
+                var message = parseMessage(children[i]);
+                if (message && (typeof message.message === "undefined"))
```

Comment:
Why are you doing nothing inside the if body?

Reply:
Sorry I was testing smth, I have removed it